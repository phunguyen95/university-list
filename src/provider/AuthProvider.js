import React, { useState, useEffect } from "react";
import { signUp, logout, signIn } from "..//firestore";
import useInputState from "../hooks/useInputState";
import useDebounce from "../hooks/useDebounce";
import _isEmpty from "lodash/isEmpty";
import { ToastContainer, toast } from "react-toastify";
import "react-toastify/dist/ReactToastify.css";
export const firebaseAuth = React.createContext();

const AuthProvider = (props) => {
  const initState = { email: "", password: "" };
  const [errors, setErrors] = useState([]);
  const [token, setToken] = useState(null);
  const [isLoading, setIsLoading] = useState(false);

  useEffect(() => {
    if (errors.length > 0) {
      toast.warn("Sorry,there's unexpected error, please try again later", {});
      setIsLoading(false);
      setErrors([]);
    }
  }, [errors]);

  const handleSignup = (email, password) => {
    // calling signup from firebase server
    signUp(email, password, setErrors, setToken);
  };
  const handleSignIn = (email, password) => {
    signIn(email, password, setErrors, setToken);
  };
  const handleSignout = () => {
    logout(setErrors, setToken);
    toast.info("You have been log out successfully", {});
  };

  return (
    <firebaseAuth.Provider
      value={{
        handleSignup,
        errors,
        token,
        handleSignout,
        handleSignIn,
        isLoading,
        setIsLoading,
        setErrors,
      }}
    >
      <ToastContainer
        position="top-right"
        autoClose={3000}
        hideProgressBar={false}
        newestOnTop={false}
        closeOnClick
        rtl={false}
        pauseOnFocusLoss
        draggable
        pauseOnHover
      />

      {props.children}
    </firebaseAuth.Provider>
  );
};
export default AuthProvider;
