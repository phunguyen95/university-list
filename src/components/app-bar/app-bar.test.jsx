import { render } from "@testing-library/react";
import AppBar from "./app-bar";
import AuthProvider from "../../provider/AuthProvider";

import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";

test("renders pagination navigaton view without crashing", () => {
    const history = createMemoryHistory();
    const handleSignout = jest.fn();

    const { asFragment } = render(
        <Router history={history}>
          <AuthProvider value={{ handleSignout }}>
              <AppBar />
          </AuthProvider>
        </Router>
      );
    
  expect(asFragment()).toMatchSnapshot();
});
