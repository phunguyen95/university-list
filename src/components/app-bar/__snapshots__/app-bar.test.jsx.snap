// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders pagination navigaton view without crashing 1`] = `
<DocumentFragment>
  <div
    class="Toastify"
  />
  <nav
    aria-label="main navigation"
    class="navbar"
    role="navigation"
  >
    <div
      class="navbar-brand"
    >
      <a
        class="navbar-item"
        href="/"
      >
        <img
          alt="Logo"
          height="28"
          src="https://bulma.io/images/bulma-logo.png"
          width="112"
        />
      </a>
      <a
        aria-expanded="false"
        aria-label="menu"
        class="navbar-burger burger "
        data-target="navbarBasicExample"
        role="button"
      >
        <span
          aria-hidden="true"
        />
        <span
          aria-hidden="true"
        />
        <span
          aria-hidden="true"
        />
      </a>
    </div>
    <div
      class="navbar-menu "
      id="navbarBasicExample"
    >
      <div
        class="navbar-start"
      >
        <div
          class="navbar-item"
        >
          <a
            class="navbar-item"
            href="/"
          >
            Home
          </a>
          <a
            class="navbar-item"
            href="/subscription"
          >
            Subscription
          </a>
          <a
            class="navbar-item"
            href="/register"
          >
            Register
          </a>
          <a
            class="navbar-item"
            href="/login"
          >
            Login
          </a>
        </div>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
